File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.java
class Solution {
    public int numSpecial(int[][] mat) {
        int count =0;
         for (int i = 0; i < mat.length; i++) {
            for (int j = 0; j <mat[0].length ; j++) {
                int c=0;
                int r=0;
                if(mat[i][j]==1){
                    for(int k=0;k<mat[0].length;k++)
                          if(mat[i][k]==1)c++;
                              for(int x=0;x<mat.length;x++)
                                  if(mat[x][j]==1)r++;
                }
                if(r==1&&c==1){ count++;}
                
            }
        }
 
          
    return count;
    }
}

File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.cpp
class Solution {
public:
    int numSpecial(vector<vector<int>>& mat) {
        vector<int>rows(mat.size(), 0), cols( mat[0].size(), 0);
        int ans = 0; 
        for(int i = 0; i < mat.size(); ++i){
            for(int j = 0; j < mat[0].size(); j++){
                if(mat[i][j]){
                    rows[i]++, cols[j]++;
                }
            }
        }

        for(int i = 0; i < mat.size(); ++i){
            for(int j = 0; j < mat[0].size(); j++){
                if(rows[i] == 1 and cols[j] == 1 and mat[i][j]){
                    ans++;
                }
            }
        }

        return ans; 
    }
};